service/history/handler/handler.go

```diff
@@ -2060,9 +2065,47 @@ func (h *handlerImpl) GetFailoverInfo(
 func (h *handlerImpl) RatelimitUpdate(
 	ctx context.Context,
 	request *types.RatelimitUpdateRequest,
-) (*types.RatelimitUpdateResponse, error) {
-	// TODO: wire up to real global-ratelimit aggregator
-	return nil, nil
+) (_ *types.RatelimitUpdateResponse, retError error) {
+	defer func() { log.CapturePanic(recover(), h.GetLogger(), &retError) }()
+	h.startWG.Wait()
+
+	scope, sw := h.startRequestProfile(ctx, metrics.HistoryRatelimitUpdateScope)
+	defer sw.Stop()
+
+	if h.isShuttingDown() {
+		return nil, constants.ErrShuttingDown
+	}
+
+	// for now there is just one ratelimit-rpc type and one algorithm that makes use of it.
+	// unpack the arg and pass it to the aggregator.
+	// in the future, this should select the algorithm by the Any.ValueType, via a registry of some kind.
+	arg, err := rpc.AnyToAggregatorUpdate(request.Any)
+	if err != nil {
+		return nil, h.error(fmt.Errorf("failed to map data to args: %w", err), scope, "", "", "")
+	}
+	err = h.ratelimitAggregator.Update(arg)
+	if err != nil {
+		return nil, h.error(fmt.Errorf("failed to update ratelimits: %w", err), scope, "", "", "")
+	}
+
+	// collect the response data and pack it into an Any for the response.
+	// like unpacking, this will eventually be handled by the registry above.
+	weights, used, err := h.ratelimitAggregator.HostWeights(arg.ID, maps.Keys(arg.Load))
+	if err != nil {
+		return nil, h.error(fmt.Errorf("failed to retrieve updated weights: %w", err), scope, "", "", "")
+	}
+	h.GetLogger().Debug("ratelimit data",
+		tag.Dynamic("used", used),
```

Comment:
Probably want to prefix these log tags aslo

Reply:
removed, this was just for local debugging purposes.  tbh I don't know how cheap these debug logs are in our setup, so I kinda want to avoid it unless we have a planned use